pyproject.toml

```diff
@@ -19,8 +19,8 @@ dependencies = [
     "openai>=1.30.1,<2.0.0",    
     "rich>=13.6.0,<15.0.0",
     "pydantic>=2.0.0, <3.0",
-    "typer>=0.9.0,<0.16",
-    "click<8.2.0", # TODO: Upgrade typer range and test cli before allowing >=8.2.0
+    "typer>=0.9.0,<0.20",
+    "click>=8.1.0,<8.4.0",
```

Comment:
Since we still support down to typer 0.9.0, the lower bounds for click should be 7.1.1

Reply:
Actually, now I think about it more, if we can support the higher versions of typer, we should be able to remove the declaration for click here as it will come as a dependency of typer.